jobs/batch_jobs/opportunity_management_jobs.py

```diff
@@ -0,0 +1,243 @@
+# coding: utf-8
+#
+# Copyright 2021 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Jobs that manage Exploration Opportunity models."""
+
+from __future__ import absolute_import
+from __future__ import annotations
+from __future__ import unicode_literals
+
+import itertools
+
+from core.domain import exp_domain
+from core.domain import exp_fetchers
+from core.domain import opportunity_services
+from core.domain import story_domain
+from core.domain import story_fetchers
+from core.domain import topic_domain
+from core.domain import topic_fetchers
+from core.platform import models
+from jobs import base_jobs
+from jobs.io import ndb_io
+from jobs.types import job_run_result
+
+import apache_beam as beam
+
+from typing import Dict, List, Union
+
+MYPY = False
+if MYPY: # pragma: no cover
+    from mypy_imports import exp_models
+    from mypy_imports import opportunity_models
+    from mypy_imports import story_models
+    from mypy_imports import topic_models
+
+(
+    exp_models, opportunity_models, story_models,
+    topic_models
+) = models.Registry.import_models([
+    models.NAMES.exploration, models.NAMES.opportunity, models.NAMES.story,
+    models.NAMES.topic
+])
+
+
+class DeleteExplorationOpportunitySummaries(base_jobs.JobBase):
+    """Job that deletes ExplorationOpportunitySummaryModels."""
+
+    def run(self) -> beam.PCollection[job_run_result.JobRunResult]:
+        """Returns a PCollection of 'SUCCESS' or 'FAILURE' results from
+        deleting ExplorationOpportunitySummaryModel.
+
+        Returns:
+            PCollection. A PCollection of 'SUCCESS' or 'FAILURE' results from
+            deleting ExplorationOpportunitySummaryModel.
+        """
+        exp_opportunity_summary_model = (
+            self.pipeline
+            | 'Get all non-deleted opportunity models' >> ndb_io.GetModels(
+                opportunity_models.ExplorationOpportunitySummaryModel.get_all(
+                    include_deleted=True))
```

Comment:
A bit confusing that this is true when the description is "non-deleted opportunity models".

Reply:
Fixed.